meet_balls_app/tests.py

```diff
@@ -67,3 +76,118 @@ def test_successful_logout(self, client, player):
+                                    client,
+                                    player,
+                                    player_rating,
+                                    court,
+                                    court_ball_game,
+                                    game_event,
+                                    game_event_player,
+                                    ):
+
+        url = reverse('profile', kwargs={'id': player.user.id})
+        client.force_login(player.user)
+        response = client.get(url)
+        assert response.status_code == 200
+        assert player.user.username in response.content.decode()
+        assert player.favorite_ball_game in response.content.decode()
+        assert str(player_rating.rating) in response.content.decode()
+        assert str(game_event.ball_game) in response.content.decode()
+
+    def test_display_edit_profile_form(self, client, player):
+        url = reverse('edit profile')
+        client.force_login(player.user)
+        response = client.get(url)
+        assert response.status_code == 200
+        assert 'Edit Player Profile' in response.content.decode()
+
+    def test_edit_profile_post_invalid(self, client, player):
+        url = reverse('edit profile')
+        client.force_login(player.user)
+        initial_birth_date = player.birth_date
+        initial_favorite_ball_game = player.favorite_ball_game
+
+        data = {
+            'birth_date': 'invalid_date',
+            'favorite_ball_game': 'Basketball',
+        }
+        assert player.favorite_ball_game != BallGame.Basketball
+        assert player.birth_date != "invalid_date"
+        response = client.post(url, data)
+        assert response.status_code == 302
+        assert response.url == reverse('edit profile')
+
+        player.refresh_from_db()
+
+        assert player.birth_date == datetime.datetime.strptime(initial_birth_date, '%Y-%m-%d').date()
+        assert player.favorite_ball_game == initial_favorite_ball_game
+
+    def test_edit_profile_post_valid(self, client, player):
+        url = reverse('edit profile')
+        client.force_login(player.user)
+        initial_birth_date = player.birth_date
+        initial_favorite_ball_game = player.favorite_ball_game
+
+        data = {
+            'birth_date': '2000-02-02',
+            'favorite_ball_game': 'Tennis',
+        }
+        expected_birth_date = datetime.datetime.strptime('2000-02-02', '%Y-%m-%d').date()
+        assert player.birth_date != expected_birth_date
+        assert player.favorite_ball_game != BallGame.Tennis
+
+        response = client.post(url, data)
+        assert response.status_code == 302
+        assert response.url == reverse('profile', kwargs={'id': player.user.id})
+
+        player.refresh_from_db()
+
+        assert player.birth_date != datetime.datetime.strptime(initial_birth_date, '%Y-%m-%d').date()
+        assert player.favorite_ball_game != initial_favorite_ball_game
```

Comment:
I think these 2 assertions are redundant. You can eliminate them by relying on the assertions in lines 181, 182, and 192, 193 instead

Reply:
fixed